src/cfgparse-global.c

```diff
@@ -1334,6 +1334,11 @@ int cfg_parse_global(const char *file, int linenum, char **args, int kwm)
 			HA_ATOMIC_STORE(&global.anon_key, tmp);
 		}
 	}
+	else if (strcmp(args[0], "mptcp") == 0) {
+		if (alertif_too_many_args(0, file, linenum, args, &err_code))
+			goto out;
+		global.tune.options |= GTUNE_MPTCP;
```

Comment:
I have another idea: what if you remove `GTUNE_MPTCP` (so here, in `global-t.h` and `cfgparse.c`), and instead, you add here:

```c
	else if (strcmp(args[0], "mptcp") == 0) {
		if (alertif_too_many_args(0, file, linenum, args, &err_code))
			goto out;
#ifdef __linux__
#ifndef IPPROTO_MPTCP
#define IPPROTO_MPTCP 262
#endif
		proto_tcpv4.sock_prot = IPPROTO_MPTCP;
		proto_tcpv6.sock_prot = IPPROTO_MPTCP;
		proto_rhttp.sock_prot = IPPROTO_MPTCP;
#else
		ha_alert("parsing [%s:%d]: '%s' is only supported on Linux.\n",
			 file, linenum, args[0]);
		err_code |= ERR_ALERT | ERR_FATAL;
		goto out;
#endif
	}
```

and in `include/haproxy/proto_rhttp.h`, adding this line after the `#include`

```c
extern struct protocol proto_rhttp;
```

By doing that, you modify the TCP (v4/v6) and the `rhttp` protocols to use `IPPROTO_MPTCP`.

I guess you will also need to modify this line: https://github.com/CrapsDorian/haproxy/blob/50d8c187423d6b7e9b1083e05370885f6d12e844/src/cfgparse.c#L1177 but I guess the CI will tell you that, e.g.:

```c
 if (proto->sock_prot != proto_tcpv4.sock_prot) {
```